nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/request/ContentLengthFilter.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.security.request;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ReadListener;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * This {@link Filter} rejects HTTP requests that exceed a specific, maximum size.
+ */
+public class ContentLengthFilter implements Filter {
+    private static final Logger logger = LoggerFactory.getLogger(ContentLengthFilter.class);
+    public final static String MAX_LENGTH_INIT_PARAM = "maxContentLength";
+    public final static int MAX_LENGTH_DEFAULT = 10_000_000;
+    private int maxContentLength;
+
+    public ContentLengthFilter() {
+        maxContentLength = MAX_LENGTH_DEFAULT;
+    }
+
+    public ContentLengthFilter(int maxLength) {
+        maxContentLength = maxLength;
+    }
+
+    @Override
+    public void init(FilterConfig config) throws ServletException {
+        String maxLength = config.getInitParameter(MAX_LENGTH_INIT_PARAM);
+        int length = maxLength == null ? MAX_LENGTH_DEFAULT : Integer.parseInt(maxLength);
+        if (length < 0) {
+            throw new ServletException("Invalid max request length.");
+        }
+        maxContentLength = length;
+        logger.info("Max content length set: " + maxLength + "b");
```

Comment:
Is the max content length in _bytes_ (`B`) or _bits_ (`b`)?

Reply:
Good catch;  updated to `B` in all cases.